fabric-mining-level-api-v1/src/testmod/java/net/fabricmc/fabric/test/mininglevel/MiningLevelTest.java

```diff
@@ -95,13 +140,28 @@ private static void test() {
 		test(errors, () -> checkMiningLevel(NEEDS_HOE, List.of(Items.WOODEN_HOE), List.of(Items.STICK)));
 		test(errors, () -> checkMiningLevel(NEEDS_SHOVEL, List.of(Items.WOODEN_SHOVEL), List.of(Items.STICK)));
 
+		test(errors, () -> checkMiningLevel(NEEDS_STONE_SHEARS, List.of(STONE_SHEARS), List.of(WOODEN_SHEARS, Items.STONE_AXE)));
+		test(errors, () -> checkMiningLevel(NEEDS_DIAMOND_SHEARS, List.of(DIAMOND_SHEARS), List.of(Items.SHEARS, STONE_SHEARS, Items.DIAMOND_PICKAXE)));
+		test(errors, () -> checkMiningLevel(NEEDS_NETHERITE_SHEARS, List.of(NETHERITE_SHEARS), List.of(DIAMOND_SHEARS, Items.SHEARS, Items.NETHERITE_SWORD)));
+
 		if (errors.isEmpty()) {
 			LOGGER.info("Mining level tests passed!");
 		} else {
 			AssertionError error = new AssertionError("Mining level tests failed!");
 			errors.forEach(error::addSuppressed);
 			throw error;
 		}
+
+		checkMiningSpeed(NEEDS_SHEARS, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
+		checkMiningSpeed(NEEDS_SHEARS_SLOW, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
+		checkMiningSpeed(NEEDS_SHEARS_FAST, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
+		checkMiningSpeed(NEEDS_STONE_SHEARS, Items.SHEARS, NETHERITE_SHEARS, Items.STICK);
+		checkMiningSpeed(NEEDS_DIAMOND_SHEARS, Items.SHEARS, NETHERITE_SHEARS, Items.STICK);
+		checkMiningSpeed(NEEDS_NETHERITE_SHEARS, Items.SHEARS, NETHERITE_SHEARS, Items.STICK);
+
+		checkMiningSpeed(Blocks.VINE, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
+		checkMiningSpeed(Blocks.WHITE_WOOL, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
+		checkMiningSpeed(Blocks.OAK_LEAVES, WOODEN_SHEARS, STONE_SHEARS, Items.SHEARS, DIAMOND_SHEARS, NETHERITE_SHEARS, GOLDEN_SHEARS, Items.STICK);
```

Comment:
I don't really see why these should be printed out. Instead, it might be useful to compare the mining speeds like the tool attribute test.

Reply:
Perhaps, I just wanted something quick and dirty to do the trick for debugging purposes and figured that plenty of other test mods spam the log with prints, so it wouldn't be too terrible.